src/diff.c

```diff
@@ -370,8 +370,7 @@ lyd_diff_userord_attrs(const struct lyd_node *first, const struct lyd_node *seco
      */
     if (*op == LYD_DIFF_OP_CREATE) {
         /* insert the instance */
-        LY_ARRAY_RESIZE_ERR_RET(schema->module->ctx, userord_item->inst, LY_ARRAY_COUNT(userord_item->inst) + 1,
```

Comment:
Because of its [description](https://github.com/CESNET/libyang/blob/libyang2/src/tree.h#L149) I always thought it an intentional feature. But yes, changing this would save many existing `NULL` checks across *libyang* and *sysrepo*. Also, there are 2 fairly macros `LY_ARRAY_RESIZE*` and `LY_ARRAY_CREATE*` so maybe keeping only one would make sense and if `LY_ARRAY_COUNT` is changed, there would be even less differences.

Reply:
Ok, I'll look at that